mreg_cli/exceptions.py

```diff
@@ -98,6 +134,30 @@ class ForceMissing(CliWarning):
     pass
 
 
+class IsNotAnIPAddress(CliWarning):
+    """Warning class for an entity that is not an IP address."""
+
+    pass
+
+
+class IsNotAnIPv4Address(CliWarning):
+    """Warning class for an entity that is not an IPv4 address."""
+
+    pass
+
+
+class IsNotAnIPv6Address(CliWarning):
+    """Warning class for an entity that is not an IPv6 address."""
+
+    pass
+
+
+class IsNotANetwork(CliWarning):
+    """Warning class for an entity that is not a network."""
+
+    pass
+
+
```

Comment:
nit: Non-standard naming scheme for these exception types

Presenting some alternatives below.

## `Invalid` prefix instead of `IsNot`

```suggestion
class InvalidIPAddress(CliWarning):
    """Warning class for an entity that is not an IP address."""

    pass


class InvalidIPv4Address(CliWarning):
    """Warning class for an entity that is not an IPv4 address."""

    pass


class InvalidIPv6Address(CliWarning):
    """Warning class for an entity that is not an IPv6 address."""

    pass


class InvalidNetwork(CliWarning):
    """Warning class for an entity that is not a network."""

    pass


```
##  `Error` suffix without `IsNot` prefix

```suggestion
class IPAddressError(CliWarning):
    """Warning class for an entity that is not an IP address."""

    pass


class IPv4AddressError(CliWarning):
    """Warning class for an entity that is not an IPv4 address."""

    pass


class IPv6Error(CliWarning):
    """Warning class for an entity that is not an IPv6 address."""

    pass


class NetworkError(CliWarning):
    """Warning class for an entity that is not a network."""

    pass


```


## `Invalid` prefix _and_ `Error` suffix

```suggestion
class InvalidIPAddressError(CliWarning):
    """Warning class for an entity that is not an IP address."""

    pass


class InvalidIPv4AddressError(CliWarning):
    """Warning class for an entity that is not an IPv4 address."""

    pass


class InvalidIPv6Error(CliWarning):
    """Warning class for an entity that is not an IPv6 address."""

    pass


class InvalidNetworkError(CliWarning):
    """Warning class for an entity that is not a network."""

    pass


```

## Original names with `Error` suffix:

```suggestion
class IsNotAnIPAddressError(CliWarning):
    """Warning class for an entity that is not an IP address."""

    pass


class IsNotAnIPv4AddressError(CliWarning):
    """Warning class for an entity that is not an IPv4 address."""

    pass


class IsNotAnIPv6AddressError(CliWarning):
    """Warning class for an entity that is not an IPv6 address."""

    pass


class IsNotANetworkError(CliWarning):
    """Warning class for an entity that is not a network."""

    pass


```